resources/lib/offline.py

```diff
@@ -7,32 +7,66 @@
     if addon.getSetting("download-folder") and xbmcvfs.exists(addon.getSetting("download-folder")):
-        choose = dialog.select(translate(32014),places)
-        if choose > -1:
-            atv_playlist = AtvPlaylist()
-            playlist_dict = atv_playlist.getPlaylistJson()
+        # Present a popup to the user and allow them to select a single location to download, or all
+        locations_chosen_index = dialog.select(translate(32014), locations)
+        if locations_chosen_index > -1:
+            # Initialize the Playlist class, and get the JSON containing URLs
+            top_level_json = AtvPlaylist().get_playlist_json()
             download_list = []
-            if playlist_dict:
-                for block in playlist_dict:
-                    for video in block['assets']:
-                        if places[choose].lower() == "all":
-                            download_list.append(video['url'])
-                        else:
-                            if places[choose].lower() == video['accessibilityLabel'].lower():
-                                download_list.append(video['url'])
-            # call downloader
+            if top_level_json:
+
+                # Parse the H264, HDR, and 4K settings to determine URL preference.
+                block_key_list = compute_block_key_list(addon.getSettingBool("enable-4k"),
+                                                        addon.getSettingBool("enable-hdr"),
+                                                        addon.getSettingBool("enable-hevc"))
+
+                # Top-level JSON has assets array, initialAssetCount, version. Inspect each block in assets
+                for block in top_level_json["assets"]:
+
+                    # If the chosen location was a specific place and not All
+                    if not locations[locations_chosen_index] == "All":
+                        # Each block contains a location whose name is stored in accessibilityLabel. These may recur
+                        # Get the location from the current block
+                        location = block["accessibilityLabel"]
+                        # Exit block processing early if the location didn't match our preference.
+                        # This prevents the location from being added to the download list
+                        if not locations[locations_chosen_index] == location:
+                            print("Current location {} is not chosen location, skipping download".format(location))
+                            continue
+
+                    # Get the URL to download
+                    url = find_ranked_key_in_dict(block, block_key_list)
+
+                    # If the URL contains HTTPS, we need revert to HTTP to avoid bad SSL cert
+                    # NOTE: Old Apple URLs were HTTP, new URLs are HTTPS with a bad cert
+                    if "https" in url:
+                        url = url.replace("https://", "http://")
```

Comment:
I had a couple different approaches to this in earlier commits, but opted for this since 1) it worked universally and 2) the plugin's existing JSON used HTTP anyway so we're at parity with the current live code. Let me know if you'd prefer a different way, but this seemed the most straightforward. 

[Before](https://github.com/enen92/screensaver.atv4/pull/47/commits/2ca1c4014fb2b2edc1ebcae2d19a650872fbf99b#diff-34b295d8a4f54f2ca324ed8dec634dc942924a50fd0b4e6195a42b4f7d710caeL134-L136) passing a URL into `self.atv4player.play(url, windowed=True)` in `atv.py` I had the following modification:

```
        # If SSL cert issues are seen and we must use HTTPS, disable SSL verification
        # https://kodi.wiki/view/SSL_certificates#Disabling_the_check
        # file = file + "|verifypeer=false"
```

This seemed to work, but only for the `xbmc.Player()`. 

If I wanted to use the HTTPS URL for grabbing `resources.tar` from Apple in `playlist.py`, I had to [use](https://github.com/enen92/screensaver.atv4/pull/47/commits/fdd8d5f938e1f7f1d925771954924683ca7f901e#diff-b55c9606829c220ce8f88091aa1c25df1701019b52a4364eafb2f5a7ddd59c67L28-L30) the following SSL inner method:

```
    # Setup for disabling SSL cert verification, as the Apple cert is bad
    # https://stackoverflow.com/questions/43204012/how-to-disable-ssl-verification-for-urlretrieve
    ssl._create_default_https_context = ssl._create_unverified_context
```

Not sure if the `|verifypeer=false` trick would work here since we're using `request.urlretrieve()` to download the tarball.

I'll leave as is with the new commit, let me know which approach you'd prefer to go forward with!